flask/test/auth_tests/test_auth.py

```diff
@@ -0,0 +1,78 @@
+import pytest
+from auth import get_user_token_string
+
+from boxwise_flask.auth_helper import (  # ,; decode_jwt,; requires_auth,; 
+    user_can_access_base,
+    authorization_test,
+    get_rsa_key,
+    get_token_from_auth_header,
+    AuthError,
+    requires_auth,
+    decode_jwt
+)
+
+def test_get_valid_jwt():
+    token = get_token_from_auth_header(get_user_token_string())
+    key = get_rsa_key(token)
+    assert key != None
+
+def test_get_invalid_jwt_no_auth_header():
+    with pytest.raises(AuthError):
+        get_token_from_auth_header(None)
+
+def test_get_invalid_jwt_no_bearer():
+    with pytest.raises(AuthError):
+        get_token_from_auth_header("no bearer")
+
+def test_get_invalid_jwt_bearer_no_token():
+    with pytest.raises(AuthError):
+        get_token_from_auth_header("bearer")
+
+def test_get_invalid_jwt_bearer_with_additonal_data():
+    with pytest.raises(AuthError):
+        get_token_from_auth_header("bearer token additional")
+
+def test_user_can_access_base_valid_user():
+    user = {
+      "base_ids": [1]
+    }
+    assert user_can_access_base(user, 1)
+
+def test_user_can_access_base_no_base_ids():
+    user = {}
+    assert not user_can_access_base(user, 1)
+
+def test_user_can_access_base_invalid_base_ids():
+    user = {
+      "base_ids": [2]
+    }
+    assert not user_can_access_base(user, 1)
+
+def test_authorization_test_bases_valid():
+    assert authorization_test("bases", base_id=1)
+
+def test_authorization_test_bases_invalid():
+    with pytest.raises(AuthError):
+        authorization_test("bases", base_id=10)
+
+def test_requires_auth_valid():
```

Comment:
I do not really get this test?
How do you pass a valid jwt token to this test?

Reply:
its done with the patch request to auth 0, I need to write one for invalid tokens but it is quite hard